### PR TITLE
Bring form-serialization into router (fixes #59, fixes #56)

### LINKED ISSUE #56: Prevent user from clicking Subscribe button twice
To prevent double charging, we should see if we can do something that prevents the user from clicking the Subscribe button twice accidentally. There could be some server logic to prevent this as well in https://github.com/mozilla/payments-service/issues/25

### LINKED ISSUE #59: Tokenise card and send to API

### PR DESCRIPTION
**Pull request by muffinresearch**
I ended up doing a lower level card data serialization so that we could have more control over the submission.

* Do the serialization with the braintree client and post to the form
* UX is me just winging it. It'll be fixed up to match the mocks in due course. (thought svg path animating that tick might be nice). 

TODOs in future branches:
* Add lots and lots of tests and error handling. :bomb: 
* Pass real data into the completion page.
* Add a button to exit the flow.

### CODE REVIEW

File: public/jsx/app.jsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ var Route = Router.Route;
 var RouteHandler = Router.RouteHandler;
 
 var CardDetails = require('views/card-details');
+var Complete = require('views/complete');
```
Comment (kumar303) [inline public/jsx/app.jsx]: nit: this makes sense as a URL but it's a little vague on its own. Maybe ``CompletePay`` and ``views/complete-pay`` ?

File: public/jsx/components/card-form.jsx [hunk 2]
```diff
@@ -54,6 +62,10 @@ module.exports = React.createClass({
     };
   },
 
+  contextTypes: {
+    router: React.PropTypes.func,
+  },
```
Comment (kumar303) [inline public/jsx/components/card-form.jsx]: This along with the mixins seem like something that would need to be in all of our components. If that's the case, could we make a base component to inherit from?

File: public/jsx/components/card-form.jsx [hunk 3]
```diff
@@ -98,6 +110,39 @@ module.exports = React.createClass({
...
+        $.ajax({
+          data: {
+            pay_method_nonce: nonce,
+            plan_id: 'mozilla-concrete-brick',
+          },
+          url: '/api/braintree/subscriptions/',
+          method: 'post',
+          dataType: 'json',
+        }).then(function(subscribeData) {
+          console.log('successfully subscribed?', subscribeData);
```
Comment (kumar303) [inline public/jsx/components/card-form.jsx]: As far as I know this response won't have any content (unless we think of something) so the log message could just be generic.

**Comment by kumar303** [comment]
Pretty slick! The direct approach with the braintree tokenize function seems like a good idea.

r+wc

File: public/jsx/app.jsx (hunk 1 above)
Comment (muffinresearch) [inline public/jsx/app.jsx]: yep, good call.

File: public/jsx/components/card-form.jsx (hunk 2 above)
Comment (muffinresearch) [inline public/jsx/components/card-form.jsx]: Good point. We should definitely consider doing something like that.

**Pull request merged**